### PR TITLE
Use enum_dispatch in aptos-faucet

### PR DESCRIPTION
**Pull request by banool**
### Description
In https://github.com/aptos-labs/aptos-core/pull/6748 @davidiw pointed out that using boxed trait objects is a bit of an antipattern with not-insignficant performance costs. I knew this was the case but the alternative was a big enum where you match on every variant for every function in your trait. Well it turns out there is a crate that removes most of the boilerplate, https://docs.rs/enum_dispatch/0.3.8/enum_dispatch/index.html. This PR changes the faucet to use that crate instead of boxed trait objects.

### Test Plan
I followed the test plan described in the [README](https://github.com/aptos-labs/aptos-core/blob/main/crates/aptos-faucet/README.md#testing).

### CODE REVIEW

**Review by JoshLind — APPROVED** [review]
Yes for enum dispatch 🥳

File: crates/aptos-faucet/core/src/funder/mod.rs [hunk 1]
```diff
@@ -4,22 +4,58 @@
...
+use async_trait::async_trait;
+use enum_dispatch::enum_dispatch;
 pub use fake::FakeFunder;
 pub use mint::MintFunder;
 use serde::{Deserialize, Serialize};
 use std::sync::Arc;
-pub use traits::Funder;
 pub use transfer::TransferFunder;
 
+/// explain
```
Comment (gregnazario) [inline crates/aptos-faucet/core/src/funder/mod.rs:25]: Lol

**Review by gregnazario — APPROVED** [review]
Just the comment that says `explain`

File: crates/aptos-faucet/core/src/funder/mod.rs (hunk 1 above)
Comment (banool) [reply crates/aptos-faucet/core/src/funder/mod.rs:25]: Broooooo I swear I fixed this lol I must not have hit save.

**Pull request merged**